File: kafka-connect-temp/src/main/scala/com/datamountaineer/streamreactor/temp/ErrorPolicySettings.scala
package com.datamountaineer.streamreactor.temp

import com.datamountaineer.streamreactor.connect.errors.{ErrorPolicy, ErrorPolicyEnum}

trait ErrorPolicySettings extends BaseSettings {
  val errorPolicyConstant: String

  def getErrorPolicy: ErrorPolicy = ErrorPolicy(ErrorPolicyEnum.withName(getString(errorPolicyConstant).toUpperCase))
}

File: java-connectors/kafka-connect-common/src/main/java/io/lenses/java/streamreactor/common/config/base/intf/ErrorPolicySettings.java
package io.lenses.java.streamreactor.common.config.base.intf;

import static io.lenses.java.streamreactor.common.config.base.constants.TraitConfigConst.MAX_RETRIES_PROP_SUFFIX;

import io.lenses.java.streamreactor.common.errors.ErrorPolicy;
import io.lenses.java.streamreactor.common.errors.ErrorPolicyEnum;

public interface ErrorPolicySettings extends BaseSettings {
  default ErrorPolicy getErrorPolicy() {
    return ErrorPolicyEnum.byName(getString(connectorPrefix() + "." + MAX_RETRIES_PROP_SUFFIX));
  }
}
